pkg/specs/property.go

```diff
@@ -83,6 +85,43 @@ type Property struct {
 	Enum      *Enum                `json:"enum,omitempty"`
 }
 
+// UnmarshalJSON corrects the 64bit data types in accordance with golang
+func (prop *Property) UnmarshalJSON(data []byte) error {
+	type property Property
+	p := property{}
+	err := json.Unmarshal(data, &p)
+	if err != nil {
+		return err
+	}
+
+	*prop = Property(p)
+	prop.Clean()
+
+	for _, nested := range prop.Nested {
+		nested.Clean()
+	}
+
+	return nil
+}
+
+// Clean fixes the type casting issue of unmarshal
+func (prop *Property) Clean() {
+	if prop.Default != nil {
+		switch prop.Type {
+		case types.Int64, types.Sint64, types.Sfixed64:
+			_, ok := prop.Default.(int64)
+			if !ok {
+				prop.Default = int64(prop.Default.(float64))
+			}
+		case types.Uint64, types.Fixed64:
+			_, ok := prop.Default.(uint64)
+			if !ok {
+				prop.Default = uint64(prop.Default.(float64))
+			}
+		}
+	}
```

Comment:
Please also handle 32 bit types

Reply:
are 32 bits also converted to float64 or float32 ?

Reply:
float32, check out the protobuf types conversion for all the available types and their go representation: https://developers.google.com/protocol-buffers/docs/proto3#scalar

Reply:
Added